### ISSUE TITLE
Possibility to reorder some user-ordered list/leaf-list entries defined in a template

### ISSUE THREAD

**Issue by QiufangMa** [netmod-wg/template-reqs#11]

**Comment by jsterne** [comment]
We'll need to carefully consider how to handle user-ordered lists, especially when multiple templates can apply to a list or leaf-list, and those templates may have different or conflicting information about the order of the entries.

**Comment by QiufangMa** [comment]
I agree we need to be careful, but I think it as a "nice-to-have" capability, in cases where the ordering carries some meaning.

If we allow multiple templates to be applied to a same node, then a priority should be specified with that, as conflict occurs if different templates declares different values for a same node. But I think the user-decided order should always takes the first priority (e.g., the entry marked with position-first should always be the first entry when merging).

**Comment by jsterne** [comment]
I'd propose we keep this simple:
1) avoid any operators like "position-first" or "position-after <x>"    (i.e. don't allow a template to specify or change order of anything)
2) for lists, the local list entries come first, then list entries are added at the end in precedence order (innermost templates first, etc)
3) for leaf-lists, just deal with entire leaf-lists as a whole entity. The local definition of the leaf-list would completely replace the entire leaf-list defined by a template  (i.e. can't mix leaf-list entries from local config and templates in a single leaf-list)
